vms/platformvm/blocks/builder/camino_builder.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2022-2023, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package builder
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/timer/mockable"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+)
+
+func getNextDeferredStakerToReward(
+	chainTimestamp time.Time,
+	shouldRewardNextCurrentStaker bool,
+	nextCurrentStakerTxID ids.ID,
+	preferredState state.Chain,
+) (ids.ID, bool, error) {
+	if !chainTimestamp.Before(mockable.MaxTime) {
+		return ids.Empty, false, errEndOfTime
+	}
+
+	deferredStakerIterator, err := preferredState.GetDeferredStakerIterator()
+	if err != nil {
+		return ids.Empty, false, err
+	}
+	defer deferredStakerIterator.Release()
+
+	if deferredStakerIterator.Next() {
+		deferredStaker := deferredStakerIterator.Value()
+		if shouldRewardNextCurrentStaker {
```

Comment:
and what if deferred staker ends earlier?

Reply:
Sure its possible, thats exactly the place where we'r on the way to reward/removal :)
And here, with `if shouldRewardNextCurrentStaker {` u just ignore deferredStaker value